src/content/components/StyleDef.js

```diff
@@ -0,0 +1,55 @@
+// inspired from https://gist.github.com/jviereck/9a71734afcfe848ddbe2 -- simplified
```

Comment:
I tried to use flow here, but it was too painful and forced me to add a lot of unnecesary if conditions... so I removed it.